pkg/platform/kube/functionres/lazy.go

```diff
@@ -1838,7 +1874,10 @@ func (lc *lazyClient) populateServiceSpec(ctx context.Context,
 	spec *v1.ServiceSpec) {
 
 	if function.Status.State == functionconfig.FunctionStateScaledToZero ||
-		function.Status.State == functionconfig.FunctionStateWaitingForScaleResourcesToZero {
+		function.Status.State == functionconfig.FunctionStateWaitingForScaleResourcesToZero ||
+		// if a function has this status, it means that it is not ready yet
+		// so we need to wait until it become ready and then update a selector
```

Comment:
A bit clearer IMO
```suggestion
		// when scaling from zero we update the service separately, only after all other resources are ready
```

Reply:
@TomerShor but we update service also before waiting for availability, so `only after` isn't really relevant. What about 

```suggestion
		// when scaling from zero we patch the service selector only after all resources are ready
```

Reply:
Yes sounds good